### PR TITLE
Fix controllers

### PR DESCRIPTION
**Pull request by atareshawty**
All of the things are broken!

### CODE REVIEW

File: controllers/Users.js [hunk 1]
```diff
@@ -21,45 +19,21 @@ function UsersController() {
...
-    promise.then((challengeableUsers) => {
-      res.render('challengeSelect', {user: req.user, challengeableUsers: challengeableUsers});
-    });
-
-    promise.catch(() => {
-      res.render('error');
-    });
-
-    return promise;
+    Promise.all([Challenge.getAllChallengeablePeopleForUser(req.user), Performance.getNext()])
```
Comment (atareshawty) [inline controllers/Users.js]: I doubt this needs to be two separate queries

File: models/Challenge.js [hunk 2]
```diff
@@ -2,6 +2,52 @@ const queries = require('./queries');
 const utils = require('../utils');
 
 class Challenge {
+  getAllChallengeablePeopleForUser(user) {
+    const sql = 'SELECT * FROM spots AS S, users AS U WHERE u.instrument = $1 AND u.part = $2 AND u.eligible = FALSE AND S.id = U.spotId';
```
Comment (atareshawty) [inline models/Challenge.js]: Maybe include performance here (or another query) so we don't have to make two queries

**Comment by atareshawty** [comment]
There might be places in this code where a model promise might not be resolving. For instance
```js
query.on('row', (result) => {
   client.end();
   resolve(result);
});
query.on('end', () => client.end());
```
If the query doesn't return any rows, the promise never gets resolved and we hang for ever 😮

File: controllers/Users.js (hunk 1 above)
Comment (atareshawty) [inline controllers/Users.js]: This definitely doesn't need to be two separate queries

File: models/User.js [hunk 3]
```diff
@@ -1,26 +1,18 @@
...
     return new Promise((resolve, reject) => {
       const client = utils.db.createClient();
-      const queryString = 'SELECT * FROM "Users" WHERE "nameNumber" = $1;';
+      const queryString = 'SELECT * FROM users WHERE nameNumber = $1;';
 
       client.connect();
       client.on('error', (err) => {reject(err);});
 
       const query = client.query(queryString, [nameNumber]);
       query.on('row', (result) => {
```
Comment (atareshawty) [inline models/User.js]: Add this. If we don't find a user, the promise won't ever resolve
```js
query.on('end', () => {
  client.end();
  resolve(null);
}
```

File: models/schema.sql [hunk 4]
```diff
@@ -109,6 +109,7 @@ CREATE TABLE users (
   eligible boolean NOT NULL DEFAULT FALSE,
 	squadLeader boolean NOT NULL DEFAULT FALSE,
 	admin boolean NOT NULL DEFAULT FALSE,
+	alternate boolean NOT NULL DEFAULT FALSE,
```
Comment (atareshawty) [inline models/schema.sql]: @btmills I suddenly feel the need to alphabetize everything

Comment (btmills) [inline models/schema.sql]: You shall be assimilated.

File: controllers/Challenges.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+  this.new = (req, res) => {
+    const userId = req.user.nameNumber,
+      spotId = req.body['challenge-form'],
+      performanceId = req.params.performanceId;
+
+    Challenge.makeChallenge(userId, spotId, performanceId)
+      .then(() => {
+        req.user.eligible = false;
+        console.log(`${req.user.name} successfully challenged for ${spotId}`); //TODO: logging
+        res.render('challengeSuccess', { user: req.user, spotId });
```
Comment (atareshawty) [inline controllers/Challenges.js]: Would we want to do a redirect to profile with a flash message?

File: models/Challenge.js [hunk 6]
```diff
@@ -1,29 +1,94 @@
-const queries = require('./queries');
 const utils = require('../utils');
 
 class Challenge {
-  makeChallenge(userId, spotId) {
-    const sql = queries.makeChallenge;
+  getAllChallengeablePeopleForUser(user) {
+    const sql = 'SELECT * FROM spots AS S, users AS U WHERE u.instrument = $1 AND u.part = $2 AND u.eligible = FALSE AND S.id = U.spotId';
```
Comment (atareshawty) [inline models/Challenge.js]:
```sql
ORDER BY S.id
```

Comment (atareshawty) [inline models/Challenge.js]: Then we're going to want to check if the spot has been challenged and add some styling it it is

File: models/Challenge.js [hunk 7]
```diff
@@ -1,29 +1,94 @@
...
   }
+
+  parseChallenge(challenge) {
+    return {
+      spotId: challenge.spotid,
+      performanceName: challenge.name
+    };
+  }
+
+  parseChallengeAblePerson(challenge) {
```
Comment (atareshawty) [inline models/Challenge.js]: Let's check if the spot has been challenged and add some styling if it is

**Pull request merged**